### ISSUE TITLE
Hunter Roam library on Arduino

### ISSUE THREAD

**Issue by deamn** [ecodina/hunter-wifi#12]
Hello, in your documentation, you wrote that it should work with Arduino board. I'm trying to use the HunterRoam library with Arduino Mega (for 5V) and Arduino Nano 33 IoT without success. Do you have some feedback with this use case? You know if a [onewire](https://www.arduino.cc/reference/en/libraries/onewire/) library is required? A particular PIN is required or D1 is ok?
With your WeMos D1 Mini Pro you don't need a common GND? Do you have another picture, because we don't see very well your board...
I have a Hunter XCore.

**Comment by ecodina** [comment]
Hi!

I do not have any of your boards, so I cannot give you specific details. However, the library should work with any Arduino-compatible board (such as my WeMos D1).

You should be able to use my complete code. Since this is a PlatformIO project, you should only have to add an environment to the [platformio.ini](https://github.com/ecodina/hunter-wifi/blob/master/esp8266-hunter-sprinkler/platformio.ini) file according to one of your boards ([Mega](https://docs.platformio.org/en/stable/boards/atmelavr/megaatmega2560.html#arduino-mega-or-mega-2560-atmega2560-mega-2560) or [Nano](https://docs.platformio.org/en/stable/boards/atmelsam/nano_33_iot.html)).

The PIN used to communicate with the Hunter Port is defined in [global_config.h](https://github.com/ecodina/hunter-wifi/blob/master/esp8266-hunter-sprinkler/include/global_config.h). In my code, I use pin 16 (D0), but any other one available should work fine.

If you only wish to use the Hunter library found in the _lib_ folder, the only Arduino-specific function used is `digitalWrite()`, so as far as you include the `Arduino.h` header, it should work. 

As for additional pictures, I am afraid I don't currently have any. I thought the diagram in the [docs ](https://github.com/ecodina/hunter-wifi/blob/master/docs/2%20Connection%20to%20the%20SmartPort.md) was enough to understand how to connect it, but I'll try to explain it here:

- You need a power source (e.g. a phone charger). This is a _floating power source_, since the voltage is not yet referenced to any common value. 

- The positive cable on the 5 V side of this power source needs to be connected to the AC#2 port. This way, the floating power supply is referenced to the voltage needed in the REM port.

I hope my explanation helps you!

**Comment by gashtaan** [comment]
I've reverse engineered the PCB and drew the schematic, maybe it will help to see it in more detail.
https://github.com/gashtaan/hunter-xcore-firmware/blob/main/schematics/controller.pdf

@ecodina I noticed in you docs that you've failed to power it up directly from 24V AC transformer. In matter of fact, you could blow MCU in Hunter controller up this way. The reason is that there is negative voltage regulation in Hunter X-Core for internal MCU (-3.3V or -5V depending on unit revision). The REM port is connected to MCU pin directly (thankfully through the resistor), therefore you need to drive it also at the same potential. And that's what you're doing when you're connecting the VCC of your board to Hunter X-Core's common ground on AC#2 terminal.

**Comment by gashtaan** [comment]
@deamn Which version of Hunter X-Core do you have? You have to look on the PCB inside.

**Comment by deamn** [comment]
@ecodina thanks for your complete answer. In between I have buy a WeMos D1 mini pro v2 and I tried with a 5V floating power source and it's work like expected. In few days I will tried with an Arduino Nano 33 IoT.
@gashtaan I don't remember how to access the PCB behind the front panel. I see that it's a model XC-401E fabrication date OCT 15

**Comment by deamn** [comment]
@ecodina hello, I bought a AC/DC CONVERTER 230VAC to 5VDC (PSK-10D-5-T from CUI inc) and the WeMos D1 mini pro v2 with your software works like expected. Thanks for the great job.

**Comment by numericOverflow** [comment]
> I've reverse engineered the PCB and drew the schematic, maybe it will help to see it in more detail. https://github.com/gashtaan/hunter-xcore-firmware/blob/main/schematics/controller.pdf
> 
> @ecodina I noticed in you docs that you've failed to power it up directly from 24V AC transformer. In matter of fact, you could blow MCU in Hunter controller up this way. The reason is that there is negative voltage regulation in Hunter X-Core for internal MCU (-3.3V or -5V depending on unit revision). The REM port is connected to MCU pin directly (thankfully through the resistor), therefore you need to drive it also at the same potential. And that's what you're doing when you're connecting the VCC of your board to Hunter X-Core's common ground on AC#2 terminal.

Awesome, so @gashtaan, can you show a pic or post a diagram of the correct connections to safely power from pins within the Hunter unit so I can get rid of my secondary power supply?

**Comment by gashtaan** [comment]
@numericOverflow no, there is no easy way to do it, there wiring is the least problem here.

**Comment by numericOverflow** [comment]
> @numericOverflow no, there is no easy way to do it, there wiring is the least problem here.

@gashtaan Ok, well good to know. I'll just keep it powered as-is then.

**Comment by rmackinnon** [comment]
@dieterrosch The AC adapter that I have for my PCC1200i shoves 12VAC over both of the two legs (AC1/AC2). But I believe a more interesting choice of circuit design has been employed on the valve controller itself that maybe keeping us from just tapping an AC leg and calling it a day.

@gashtaan 's schematic doesn't fully map up to my facepack and valve control boards. But upon inspection, I noticed on my valve control board (labeled PCC+ 12S v1.11) has AC1 and GND bridged together, where as AC2 is not directly connected.  The AC2 circuit network bridges to AC1 via Hunaan VDK-10D560K varistor (RV14), hits a chonky 5W 0.82Ω dump resistor (R51), then splits with one leg going to a 50V 220µF cap (C16) and then a power rail for the valve controls FETs (Q8-20), and the other leg to the GND pin on an ST L7905CV for negative voltage regulation (U2).

**Comment by gashtaan** [comment]
@rmackinnon I reverse engineered X-Core units, never got my hand on any Pro-C unit, so I can't help much, but from what you said, the design seems to be similar. Only for the sake of curiosity, can you look what microcontroller they use here?

**Comment by rmackinnon** [comment]
@gashtaan The MCU used has a decent amount of epoxy for IP6x waterproofing. But taking a look at the pinout for the PIC18F86K90, all of the pins line up.  So my best guess is that it's a PIC18F8x compatible. Given some time I'll setup a repo to do some simple KiCAD schematic.  One of the more interesting features of the Pro-C and ICC units and  Hydrawise facepacks is what I call a pickle plate that plugs into valve control board 2x10 pin cable port that normally runs straight to the facepack and offers an RJ45 for connection to the WIFI and LAN modules for remote management.

**Comment by gashtaan** [comment]
@rmackinnon I'm looking forward to a schematic from you. In X-Core the voltage regulation is horribly inefficient and as you mentioned 7905 regulator in Pro-C unit, they most likely stick to old design. I'm curious how they handle power to those external modules.

**Comment by rmackinnon** [comment]
Moved conversation to [here](https://github.com/rmackinnon/hunter-pcc1200/issues/1).
